kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -947,19 +948,67 @@ protected void handleFindCoordinatorRequest(KafkaHeaderAndRequest findCoordinato
                 });
     }
 
+    private <T> void replaceTopicPartition(Map<TopicPartition, T> replacedMap,
+                                           Map<TopicPartition, TopicPartition> replacingIndex) {
+        Map<TopicPartition, T> newMap = new HashMap<>();
+        replacedMap.entrySet().removeIf(entry -> {
+            if (replacingIndex.containsKey(entry.getKey())) {
+                newMap.put(replacingIndex.get(entry.getKey()), entry.getValue());
+                return true;
+            }
+            return false;
+        });
+        replacedMap.putAll(newMap);
+    }
+
     protected void handleOffsetFetchRequest(KafkaHeaderAndRequest offsetFetch,
                                             CompletableFuture<AbstractResponse> resultFuture) {
         checkArgument(offsetFetch.getRequest() instanceof OffsetFetchRequest);
         OffsetFetchRequest request = (OffsetFetchRequest) offsetFetch.getRequest();
         checkState(groupCoordinator != null,
             "Group Coordinator not started");
 
+        Map<TopicPartition, OffsetFetchResponse.PartitionData> unknownPartitions = new HashMap<>();
+
+        // replace
+        Map<TopicPartition, TopicPartition> replacingIndex = new HashMap<>();
         KeyValue<Errors, Map<TopicPartition, OffsetFetchResponse.PartitionData>> keyValue =
             groupCoordinator.handleFetchOffsets(
-                request.groupId(),
-                Optional.ofNullable(request.partitions())
+                    request.groupId(),
+                    request.partitions() == null ? Optional.ofNullable(request.partitions()) : Optional.of(
+                            request.partitions()
+                            .stream()
+                            .map(tp -> {
+                                try {
+                                    TopicPartition newTopicPartition = new TopicPartition(
+                                            new KopTopic(tp.topic()).getFullName(), tp.partition());
+                                    replacingIndex.put(newTopicPartition, tp);
+                                    return newTopicPartition;
+                                } catch (Exception e) {
```

Comment:
Catch `RuntimeException` here. Though I think a better fix is to just throw `IllegalArgumentException` in `KopTopic`'s constructor and catch `IllegalArgumentException` here.